src/briefcase/integrations/subprocess.py

```diff
@@ -26,6 +31,53 @@ def json_parser(json_output):
         raise ParseError(f"Failed to parse output as JSON: {e}")
 
 
+def is_process_dead(pid: int):
+    """
+    Returns True if a PID is not assigned to a process.
+
+    Checking if a PID exists is only a semi-safe proxy to determine
+    if a process is dead since PIDs can be re-used. Therefore, this
+    function should only be used via constant monitoring of a PID
+    to identify when the process goes from existing to not existing.
+
+    :param pid: integer value to be checked if assigned as a PID.
+    :return: True if PID doesn't exist; defaults to False if PID=None.
+    """
+    if pid is None:
+        return False
+    return not psutil.pid_exists(pid)
+
+
+def get_process_id_by_command(command_list: list = None, command: str = ""):
+    """
+    Find a Process ID (PID) a by its command.
+    If multiple processes are found, then the most recently created
+    process ID is returned.
+
+    :param command_list: list of a command's fully qualified path and its arguments.
+    :param command: a partial or complete fully-qualified filepath to a command.
+        This is primarily intended for use on macOS where the `open` command
+        takes a filepath to a directory for an application; therefore, the actual
+        running process will be running a command within that directory.
+    :return: PID if found else None
+    """
+    matching_processes = []
+    for proc in psutil.process_iter(["cmdline", "create_time", "pid"]):
```

Comment:
Would be worth a small docstring here to clarify the purpose of this line - retrieving the cmdline, creation time and PID of all currently running processes.

Reply:
Updated.